### ISSUE TITLE
Bug: nats_jetstream input has a logic error in the stream check

### ISSUE THREAD

**Issue by maxarndt** [redpanda-data/connect#2576]
In nats_jetstream input processor, I cannot use the `stream` option without `subject` set.

Acording to the [docs](https://www.benthos.dev/docs/components/inputs/nats_jetstream#stream) this should work:

https://github.com/benthosdev/benthos/blob/88c01d179d3aa4030795ac0f52459dd7a333284b/website/docs/components/inputs/nats_jetstream.md?plain=1#L190-L192

## Reproduction

```yaml
input:
  nats_jetstream:
    urls: [nats://srv:port]
    stream: "somestream"
```

Error:
> ERRO Service closing due to: failed to init input <no label> path root.input: subject is empty  @service=benthos

## Solution

I'm going to open a PR. Hope this helps.

Edit: here it is #2577

**Comment by josephwoodward** [comment]
@mihaitodor, I believe this issue can be closed now?

**Comment by mihaitodor** [comment]
Yep, the fix made it in [v4.29.0](https://github.com/redpanda-data/connect/releases/tag/v4.29.0).
